ast/src/main/scala/jawn/ast/JValue.scala

```diff
@@ -143,7 +145,7 @@ object JNum { self =>
 
   final def hybridEq(x: Long, y: Double): Boolean = {
     val z = x.toDouble
-    y == z && z.toLong == x
+    y == z && y.toLong == x
```

Comment:
@non thx for the clear explanation! I understand now. do you want me to revert my change then?